elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchIndexManagerFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.impl;
+
+import org.hibernate.search.engine.impl.DefaultIndexManagerFactory;
+import org.hibernate.search.indexes.spi.IndexManager;
+
+/**
+ * Index manager capable of resolving the "elasticsearch" alias. All other names/aliases will be handled by the default
+ * implementation.
+ * <p>
+ * Registered through the Java service loader mechanism.
+ *
+ * @author Gunnar Morling
+ */
+public class ElasticsearchIndexManagerFactory extends DefaultIndexManagerFactory {
```

Comment:
Man it's tempting as it's a nice short and clean PR, but this would make all the Elasticsearch work completely unusable for Infinispan Query!

The services system isn't designed for this as it would blow up if it were to find multiple alternative implementations; Infinispan Query also overrides the DefaultIndexManagerFactory to add its own short names.

We need to create a "short name contributors" concepts and be able to apply all of them, not picking only one.